### PR TITLE
PR for enhanced data portal crawling

### PR DESCRIPTION
**Pull request by sritejakv**
PR for enhanced data portal crawling

### CODE REVIEW

File: squirrel.worker/src/main/java/org/dice_research/squirrel/analyzer/impl/html/scraper/HtmlScraper.java [hunk 1]
```diff
@@ -219,47 +175,95 @@ public HtmlScraper() {
...
+        this.uri = curi.getUri().toString();
+        Set<Triple> triples = new LinkedHashSet<Triple>();
+        this.label = uri.substring(uri.lastIndexOf("/")+1, uri.length());
 
-//        if(uri.contains("?")) {
-//         this.label = uri.substring(uri.lastIndexOf("/")+1, uri.lastIndexOf("?"));
-//        }else {
-//         this.label = uri.substring(uri.lastIndexOf("/")+1, uri.length());
-//        }
+        if (!htmlFile.toString().contains("test")) //To prevent downloading a page when running unit test cases
```
Comment (MichaelRoeder) [inline squirrel.worker/src/main/java/org/dice_research/squirrel/analyzer/impl/html/scraper/HtmlScraper.java]: Please find a better way to handle these cases. Simply checking for "test" in a URL is a little bit too error prone.

File: squirrel.worker/src/main/java/org/dice_research/squirrel/analyzer/impl/html/scraper/HtmlScraper.java [hunk 2]
```diff
@@ -219,47 +175,95 @@ public HtmlScraper() {
...
+    }
 
-        Set<Triple> triples = new LinkedHashSet<Triple>();
+    /**
+     * Method to execute java script commands in a html page.
+     * @param uri
+     * @throws IOException
+     */
+    private void executeJavaScript(CrawleableUri uri){
+        WebClient webClient = new WebClient(BrowserVersion.FIREFOX_60);
```
Comment (MichaelRoeder) [inline squirrel.worker/src/main/java/org/dice_research/squirrel/analyzer/impl/html/scraper/HtmlScraper.java]: Please check whether you can reuse the `WebClient` object. At the moment, the single methods are creating new instances all the time which may waste a lot of resources.

File: squirrel.worker/src/main/java/org/dice_research/squirrel/analyzer/impl/html/scraper/HtmlScraper.java [hunk 3]
```diff
@@ -324,15 +328,12 @@ private String replaceCommands(String s) {
     	
     	List<Node> listNodes = new ArrayList<Node>();
     	
-    	@SuppressWarnings("unused")
-		boolean useResource = false;
-    	
     	 try {
           	
-          	if(cssQuery.startsWith("l(")) {
```
Comment (MichaelRoeder) [inline squirrel.worker/src/main/java/org/dice_research/squirrel/analyzer/impl/html/scraper/HtmlScraper.java]: The bracket had to be there. Why did you remove it?

File: squirrel.worker/src/main/java/org/dice_research/squirrel/analyzer/impl/html/scraper/HtmlScraper.java [hunk 4]
```diff
@@ -66,59 +59,19 @@ public HtmlScraper() {
...
+        uri= curi.getUri().toString();
         YamlFile yamlFile = (YamlFile) yamlFiles.get(UriUtils.getDomainName(uri)).clone();
-        
-        
-        this.uri = uri;
-	     if(uri.contains("?")) {
-	      this.label = uri.substring(uri.lastIndexOf("/")+1, uri.lastIndexOf("?"));
-	     }else {
-	      this.label = uri.substring(uri.lastIndexOf("/")+1, uri.length());
-	     }
```
Comment (MichaelRoeder) [inline squirrel.worker/src/main/java/org/dice_research/squirrel/analyzer/impl/html/scraper/HtmlScraper.java]: Why did you remove this code block?

File: squirrel.worker/src/main/java/org/dice_research/squirrel/analyzer/impl/html/scraper/HtmlScraper.java [hunk 5]
```diff
@@ -324,15 +328,12 @@ private String replaceCommands(String s) {
...
-		boolean useResource = false;
-    	
     	 try {
           	
-          	if(cssQuery.startsWith("l(")) {
+          	if(cssQuery.startsWith("l")) {
           		
           		String val = cssQuery.substring(cssQuery.indexOf("(")+1,cssQuery.lastIndexOf(")"));
-//          		String label = uri.substring(uri.lastIndexOf("/")+1, uri.lastIndexOf("?"));
+          		String label = uri.substring(uri.lastIndexOf("/")+1, uri.length());
```
Comment (MichaelRoeder) [inline squirrel.worker/src/main/java/org/dice_research/squirrel/analyzer/impl/html/scraper/HtmlScraper.java]: Defining the label here is ok. However, why do you remove the search for `?` ?

File: squirrel.worker/src/main/java/org/dice_research/squirrel/analyzer/impl/html/scraper/HtmlScraper.java [hunk 6]
```diff
@@ -349,12 +350,7 @@ private String replaceCommands(String s) {
               	arrayElements[0] = el;
               	elements = new Elements(arrayElements); 
               }else {
-            	  
-            	  if(cssQuery.startsWith("res(")) {
-            		  useResource = true;
-            		  cssQuery = cssQuery.substring(cssQuery.indexOf("(")+1, cssQuery.lastIndexOf(")") );
-            	  }
```
Comment (MichaelRoeder) [inline squirrel.worker/src/main/java/org/dice_research/squirrel/analyzer/impl/html/scraper/HtmlScraper.java]: Why did you remove this `if` code block?

File: squirrel.worker/src/test/java/org/dice_research/squirrel/analyzer/impl/data_portals/CatalogDataGov.java [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+    public static Collection<Object[]> data() throws URISyntaxException {
+        List<Object[]> testConfigs = new ArrayList<Object[]>();
+        String uriDetailsPage = "https://catalog.data.gov/dataset/demographic-statistics-by-zip-code-acfc9";
+        testConfigs.add(new Object[]{
+            new CrawleableUri(new URI(uriDetailsPage)),
+            new File("src/test/resources/html_scraper_analyzer/catalog_data_gov/details_page.html"),
+            ModelFactory.createDefaultModel().add(
+                new StatementImpl(
+                    new ResourceImpl(uriDetailsPage),
+                    new PropertyImpl(DCTerms.title.toString()),
```
Comment (MichaelRoeder) [inline squirrel.worker/src/test/java/org/dice_research/squirrel/analyzer/impl/data_portals/CatalogDataGov.java]: You can use
```java
new StatementImpl(
                    ...,
                    DCTerms.title,
```

**Comment by MichaelRoeder** [comment]
Please note that the implementation has failing test cases. You can see them in Travis, e.g., https://travis-ci.org/dice-group/Squirrel/builds/543968487#L7922

File: squirrel.worker/src/main/java/org/dice_research/squirrel/analyzer/impl/html/scraper/HtmlScraper.java (hunk 3 above)
Comment (sritejakv) [reply squirrel.worker/src/main/java/org/dice_research/squirrel/analyzer/impl/html/scraper/HtmlScraper.java]: Added it back. It was missed during resolving merge conflicts.

File: squirrel.worker/src/main/java/org/dice_research/squirrel/analyzer/impl/html/scraper/HtmlScraper.java (hunk 4 above)
Comment (sritejakv) [reply squirrel.worker/src/main/java/org/dice_research/squirrel/analyzer/impl/html/scraper/HtmlScraper.java]: Added it back. It was missed during resolving merge conflicts.

File: squirrel.worker/src/main/java/org/dice_research/squirrel/analyzer/impl/html/scraper/HtmlScraper.java (hunk 6 above)
Comment (sritejakv) [reply squirrel.worker/src/main/java/org/dice_research/squirrel/analyzer/impl/html/scraper/HtmlScraper.java]: Added it back. It was missed during resolving merge conflicts.

File: squirrel.worker/src/main/java/org/dice_research/squirrel/analyzer/impl/html/scraper/HtmlScraper.java (hunk 5 above)
Comment (sritejakv) [reply squirrel.worker/src/main/java/org/dice_research/squirrel/analyzer/impl/html/scraper/HtmlScraper.java]: Removed this line as the label is being initialized in scrape method.

File: squirrel.worker/src/test/java/org/dice_research/squirrel/analyzer/impl/data_portals/CatalogDataGov.java (hunk 7 above)
Comment (sritejakv) [reply squirrel.worker/src/test/java/org/dice_research/squirrel/analyzer/impl/data_portals/CatalogDataGov.java]: Updated the same in all test cases.

**Comment by sritejakv** [comment]
> Please remove the files
> 
> ```
> spring-config/context-sparql.xml
> spring-config/context.xml
> spring-config/default-config.xml
> ```
> 
> since they are outdated. Please make sure that you use the same files as in https://github.com/dice-group/Squirrel/tree/develop/spring-config
> 
> We introduced an .editorconfig file (see https://github.com/dice-group/Squirrel/blob/develop/.editorconfig). Please copy it to your project and configure your IDE to use it. This should fix most of the changes which were introduced because of different formatting.
